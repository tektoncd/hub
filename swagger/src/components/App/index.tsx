import React from 'react';
import SwaggerUI from 'swagger-ui-react';
import 'swagger-ui-react/swagger-ui.css';
import { API_URL } from '../../config/constants';

const App: React.FC = () => {
  return (
    <div>
      <SwaggerUI url={API_URL} />
    </div>
  );
};

export default App;
