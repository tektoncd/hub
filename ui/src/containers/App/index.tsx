import React from 'react';
import { observer } from 'mobx-react';
import '@patternfly/react-core/dist/styles/base.css';
import { Page } from '@patternfly/react-core';
import { BrowserRouter as Router, Route, Routes, Navigate } from 'react-router-dom';
import HomePage from '../../components/HomePage';
import ResourceDetailsPage from '../../components/ResourceDetailsPage';
import Header from '../../containers/Header';
import Footer from '../../components/Footer';
import ParseUrl from '../ParseUrl';
import { createProvider } from '../../store/root';
import './App.css';

const Provider = createProvider();

const App: React.FC = observer(() => {
  return (
    <Provider>
      <Router>
        <ParseUrl />
        <Page header={<Header />} className="hub-page">
          <Routes>
            <Route path="/login" element={<Navigate to="/" replace />} />
            <Route path="/" element={<HomePage />}></Route>
            <Route path="/:catalog/:kind/:name/:version?" element={<ResourceDetailsPage />}></Route>
          </Routes>
          <Footer />
        </Page>
      </Router>
    </Provider>
  );
});

export default App;
