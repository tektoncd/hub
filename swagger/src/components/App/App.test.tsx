import React from 'react';
import { shallow } from 'enzyme';
import App from '.';

test('renders learn react link', () => {
  const app = shallow(<App />);
  expect(app.debug()).toMatchSnapshot();
});
