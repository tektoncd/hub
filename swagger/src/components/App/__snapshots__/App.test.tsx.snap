// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `"<div />"`;
